webrender/src/tiling.rs

```diff
@@ -718,23 +836,72 @@ impl RenderTask {
         }
     }
 
-    fn new_prim_cache(cache_info: &PrimitiveCacheInfo,
+    pub fn new_prim_cache(key: PrimitiveCacheKey,
+                      size: DeviceSize,
                       prim_index: PrimitiveIndex) -> RenderTask {
         RenderTask {
-            id: RenderTaskId::Dynamic(RenderTaskKey::CachePrimitive(cache_info.key)),
+            id: RenderTaskId::Dynamic(RenderTaskKey::CachePrimitive(key)),
             children: Vec::new(),
-            location: RenderTaskLocation::Dynamic(None, cache_info.size),
+            location: RenderTaskLocation::Dynamic(None, size),
             kind: RenderTaskKind::CachePrimitive(prim_index),
         }
     }
 
+    // Construct a render task to apply a blur to a primitive. For now,
+    // this is only used for text runs, but we can probably extend this
+    // to handle general blurs to any render task in the future.
+    // The render task chain that is constructed looks like:
+    //
+    //    PrimitiveCacheTask: Draw the text run.
+    //           ^
+    //           |
+    //    VerticalBlurTask: Apply the separable vertical blur to the primitive.
+    //           ^
+    //           |
+    //    HorizontalBlurTask: Apply the separable horizontal blur to the vertical blur.
+    //           |
+    //           +---- This is stored as the input task to the primitive shader.
+    //
+    pub fn new_blur(key: PrimitiveCacheKey,
+                    size: DeviceSize,
+                    blur_radius: DeviceLength,
+                    prim_index: PrimitiveIndex) -> RenderTask {
+        let prim_cache_task = RenderTask::new_prim_cache(key,
+                                                         size,
+                                                         prim_index);
+
+        let blur_target_size = size + DeviceSize::new(2 * blur_radius.0,
+                                                      2 * blur_radius.0);
+
+        let blur_task_v = RenderTask {
+            id: RenderTaskId::Dynamic(RenderTaskKey::VerticalBlur(blur_radius.0, prim_index)),
+            children: vec![prim_cache_task],
+            location: RenderTaskLocation::Dynamic(None, blur_target_size),
+            kind: RenderTaskKind::VerticalBlur(blur_radius, prim_index),
```

Comment:
Yea - it makes it a bit easier to extract the info (since you don't have to match on all the key kinds). But I do agree it's a bit redundant and we might want to consider changing how this works in the future.